docs/source-app/core_api/lightning_app/dynamic_work_content.rst

```diff
@@ -46,6 +46,25 @@ There are a couple of ways you can add a dynamic Work:
 	        # Run the `Work` component.
             getattr(self,  "work").run()
 
+.. note:: 
+     ``__setattr__()`` has to be triggered in order for the state to be replicated.
+
+     Currently a state update is triggered if the ``__setattr__()`` function is called. For example, if you use ``status.run_script = True``,
+     then ``__setattr__()`` is called and the status is updated with all found changes.
+     But if you just use ``state.form_values["script_dir"]  = st_script_dir`` then ``__setattr__()`` is not called
+     so you only get attributes and assign values to the value retrieved (not the state).
+     So that assignemnt would not update the state and would not be visible in other components.
```

Comment:
Where is this information from? This has been addressed a long time ago [1, 2] (pre-launch) by me and Thomas. If there is evidence of a regression, we need to fix it instead. 

[1] https://linear.app/lightning-ai/issue/LAI2-8436/work-state-shared-stated-doesnt-get-updated-for-in-place-operations
[2] https://github.com/Lightning-AI/lightning-app/pull/610

Reply:
@awaelchli 
Looks like this was reopened yesterday: https://linear.app/lightning-ai/issue/LAI2-9676/app-stops-with-error-in-lightningwork-attributeerror-list-object-has

Reply:
Thanks. Will wait for the reply on the linear ticket :)